qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -14,233 +14,393 @@
 
 # pylint: disable=invalid-name
 """
-Decompose single-qubit unitary into Euler angles.
+Decompose a single-qubit unitary via Euler angles.
 """
 
 import math
 import numpy as np
 import scipy.linalg as la
 
 from qiskit.circuit.quantumcircuit import QuantumCircuit
-from qiskit.extensions.standard import HGate, U3Gate, U1Gate, RXGate, RYGate, RZGate
+from qiskit.extensions.standard import (U3Gate, U1Gate, RXGate, RYGate, RZGate,
+                                        RGate)
 from qiskit.exceptions import QiskitError
-from qiskit.quantum_info.operators import Operator
 from qiskit.quantum_info.operators.predicates import is_unitary_matrix
 
-DEFAULT_ATOL = 1e-12
+DEFAULT_SIMPLIFY_TOLERANCE = 1e-12
+DEFAULT_ATOL = 1e-7
```

Comment:
did something happen making it necessary to make tolerance more relaxed? If you look at `test_synthesis.py` there are some tests for "hard_thetas" that explicitly change tolerance from `1e-12` to `1e-7`. If you change the global one here, then those are not necessary.

Reply:
I changed it to use two different tolerance values: one for checking when to drop unnecessary gates close to identity based on theta values (DEFAULT_SIMPLIFY_TOLERANCE) and a more relaxed one for numerical comparison of matrices for checking synethsised matrices are equivalent (DEFAULT_ATOL). Different values for both of these can also be set the class init method if you don't want to use the default values.

Previously both were using the same value but this ran into problem: You need a strict default tolerance for dropping gates or the hard theta tests will all fail, so that remains at 1e-12. But using that value for checking equivalence of final matrix I was getting constant failures for some decompositions due to imprecision of numpy multiplication

Reply:
Ok sounds good. I think you can just have one tolerance here if you move the validation into tests.